src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -514,56 +513,60 @@ exprt string_constraint_generatort::add_axioms_for_correct_number_format(
 }
 
 /// add axioms corresponding to the Integer.parseInt java function
-/// \par parameters: function application with one string expression
+/// \param f: a function application with either one string expression or one
+///   string expression and an expression for the radix
 /// \return an integer expression
 exprt string_constraint_generatort::add_axioms_for_parse_int(
   const function_application_exprt &f)
 {
-  string_exprt str=get_string_expr(args(f, 1)[0]);
+  PRECONDITION(f.arguments().size()==1 || f.arguments().size()==2);
+  string_exprt str=get_string_expr(f.arguments()[0]);
+  const exprt radix=
+    f.arguments().size()==1?from_integer(10, f.type()):f.arguments()[1];
```

Comment:
Somewhat similarly to @smowton I inferred that we should use spaces with ternary from the passage that says that we should use spaces around logical operators (I see ternary as logical operator). In the code that @owen-jones-diffblue linked there is a line break after ? so it is not that bad.